src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1139,14 +1139,45 @@ public static bool IsParameterTypeContext(this SyntaxTree syntaxTree, int positi
         {
             tokenOnLeftOfPosition = tokenOnLeftOfPosition.GetPreviousTokenIfTouchingWord(position);
 
-            if (tokenOnLeftOfPosition.IsKind(SyntaxKind.OpenParenToken))
+            if (tokenOnLeftOfPosition.Parent?.IsKind(SyntaxKind.ParenthesizedExpression,
+                    SyntaxKind.TupleExpression, SyntaxKind.TupleType) == true ||
+                tokenOnLeftOfPosition.Parent?.Parent?.IsKind(SyntaxKind.ParenthesizedExpression,
+                    SyntaxKind.TupleExpression, SyntaxKind.TupleType) == true)
             {
-                return tokenOnLeftOfPosition.Parent.IsKind(SyntaxKind.ParenthesizedExpression,
-                    SyntaxKind.TupleExpression, SyntaxKind.TupleType);
+                return true;
+            }
+
+            // (int a, i$$
+            // ($$int a, int b
+            if (tokenOnLeftOfPosition.Parent?.IsKind(SyntaxKind.ParameterList) == true &&
+                tokenOnLeftOfPosition.Parent?.Parent?.IsKind(SyntaxKind.ParenthesizedLambdaExpression) == true)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public static bool IsPossibleDeconstructionDesignation(this SyntaxTree syntaxTree,
+            int position, CancellationToken cancellationToken)
+        {
+            var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
```

Comment:
can you keep this consistent and call ".GetPreviousTokenIfTouchingWord(position);"  i have a feeling we'll have  bug (at least in the future) if you dont.